src/util/StringUtils.h

```diff
@@ -244,6 +260,35 @@ constexpr bool constantTimeEquals(std::string_view view1,
   return impl(toVolatile(view1), toVolatile(view2));
 }
 
+// _________________________________________________________________________
+template <typename Range, typename TranslationFunction>
+static std::string listToString(Range&& r,
+                                TranslationFunction translationFunction,
+                                std::string_view listItemSeparator) {
```

Comment:
In general I like this interface, I have the following suggestions:
* call it `lazyStrJoin` (it is a `StrJoin` that is `lazy` in the sense that it accepts lazy ranges).
* Add requirements that `Range` is an `std::ranges::input_range`, and that you can call the Translation function on the range elements and get something that is streamable.
* Make the translation function the last argument and default it to `std::identity`.
*  call the separator just `separator`.
* Create an second overload that takes and returns a `std::ostream&` as first argument (when we write to streams anyway), and use this to implement the version that returns a `string`.

Reply:
I agree with most of your point, but not the first one. A function name should describe a function and `StrJoin` sounds more like a function, that concatenates string.
Additionaly, what do you mean with lazy ranges? I thought only views were lazy?

Reply:
New thought, after thinking about

> Create an second overload that takes and returns a std::ostream& as first argument (when we write to streams anyway), and use this to implement the version that returns a string.

for a while. What point is there in returning a `std::ostream`, that the user **gives** us? Why not just return void? It would have the same effect...

Reply:
Implemented a version of this in [a7b6b1b](https://github.com/ad-freiburg/qlever/pull/1010/commits/a7b6b1b1f319a4afceff29fc2f520d9d9a75b724). We can discuss the points, that I raised.

Reply:
Concerning my points that you wanted to discuss about:

2. (easy) You are right, the function should return void. `operator<<` should return the `ostream&`, but here that is not really necessar, I mixed something up.
1. The name depends. In my point of view, this function concatenates a list of strings, and inbetween puts a user-defined separator. This function is called `join` in python, `StrJoin` in `absl` and `join_with` in C++23 (the last one not only for strings). It does so in a lazy fashion, that means that the inputs can be lazily created by e.g. an `input_range` and don't have to be materialized in advance. The part `lazy` was important for me, to see the difference to e.g. `absl::StrJoin`. The (optional) transformation function to convert the elements is in my understanding not the important part. It is convenient, but you could also perform the transformation on the input before passing it to the function. This naming scheme would be consistent with other function from `absl` or `std::ranges` where the name says `what do we actually do` and the optional transformations/projections are just optionally there, but don't reflect in the name.

Reply:
Ah, that makes more sense. Thank you. Done in [febf8e2](https://github.com/ad-freiburg/qlever/pull/1010/commits/febf8e2d835b560a4a354208adce48bd40e03132).